src/model/formats/foreflight-utils.ts

```diff
@@ -0,0 +1,93 @@
+import { ChecklistGroup_Category } from '../../../gen/ts/checklist';
+import { v4 as uuid } from 'uuid';
+import { ForeFlightFormatError } from './foreflight-format';
+
+export enum ForeFlightCategory {
+  Normal = 'normal',
+  Abnormal = 'abnormal',
+  Emergency = 'emergency',
+}
+
+enum KeyUsage {
+  Encrypt = 'encrypt',
+  Decrypt = 'decrypt',
+}
+
+export class ForeFlightUtils {
+  public static readonly FILE_EXTENSION = 'fmd';
+
+  public static readonly CONTAINER_TYPE = 'checklist';
+  public static readonly SCHEMA_VERSION = '1.0';
+
+  public static readonly ITEM_HEADER = 'comment';
+  public static readonly NOTE_INDENT = 1;
+
+  public static readonly OBJECT_ID_LENGTH = 32;
+
+  static readonly CIPHER_TYPE = 'AES-CBC';
+  static readonly CIPHER_BLOCK_SIZE = 16;
+  static readonly CIPHER_KEY = Buffer.from('81e06e41a93f3848', 'ascii');
+
+  static readonly encoder = new TextEncoder();
+  static readonly decoder = new TextDecoder();
+
+  private static getKey(keyUsage: KeyUsage): Promise<CryptoKey> {
+    return window.crypto.subtle.importKey(
+      'raw',
+      ForeFlightUtils.CIPHER_KEY,
+      { name: ForeFlightUtils.CIPHER_TYPE },
+      false,
+      [keyUsage],
+    );
+  }
+
+  public static async decrypt(stream: ArrayBuffer): Promise<string> {
+    const decrypted = await window.crypto.subtle.decrypt(
+      {
+        name: ForeFlightUtils.CIPHER_TYPE,
+        iv: new Uint8Array(stream.slice(0, ForeFlightUtils.CIPHER_BLOCK_SIZE)),
+      },
+      await ForeFlightUtils.getKey(KeyUsage.Decrypt),
+      new Uint8Array(stream.slice(ForeFlightUtils.CIPHER_BLOCK_SIZE)),
+    );
+    return ForeFlightUtils.decoder.decode(new Uint8Array(decrypted));
+  }
+
+  public static async encrypt(data: string): Promise<Blob> {
+    const iv = window.crypto.getRandomValues(new Uint8Array(ForeFlightUtils.CIPHER_BLOCK_SIZE));
+    return new Blob(
+      [
+        iv,
+        await window.crypto.subtle.encrypt(
+          { name: ForeFlightUtils.CIPHER_TYPE, iv: iv },
+          await ForeFlightUtils.getKey(KeyUsage.Encrypt),
+          ForeFlightUtils.encoder.encode(data),
+        ),
+      ],
+      { type: 'application/octet-stream' },
+    );
+  }
+
+  public static categoryToEFIS(category: string): ChecklistGroup_Category {
+    switch (category) {
+      case ForeFlightCategory.Normal:
+        return ChecklistGroup_Category.NORMAL;
+      case ForeFlightCategory.Abnormal:
+        return ChecklistGroup_Category.ABNORMAL;
+      case ForeFlightCategory.Emergency:
+        return ChecklistGroup_Category.EMERGENCY;
+      default:
+        throw new ForeFlightFormatError(`unknown category '${category}'`);
+    }
+  }
+
+  public static async getObjectId(): Promise<string> {
+    const hash = Array.from(
+      new Uint8Array(await window.crypto.subtle.digest('SHA-256', ForeFlightUtils.encoder.encode(uuid()))),
```

Comment:
If the Object ID is NOT in a specific format, any reason not to just take the UUID (or part of it) and use it directly without hashing? the UUID is already guaranteed unique.

A consequence would be that you can get rid of most async/Promise handling in the writing code and make it simpler.

Reply:
That was a majestic brain fart, thanks for speaking up! Fixed in dd1acba.

Obviously, I don't have any official information on what `objectId` is and how it is used, but my guess is that it is used in some way to synchronize checklists between devices. I made the following observations:

1. It always seems to be 32 characters long.
2. It always seems to contain characters from `[a-f0-9]`.
3. Objects with the same properties have different `objectId`.

It looked very much like 128 bits of "something" encoded as hex, but that "something" somehow doesn't seem to be a hash of the object. So, to try to keep our files as indistinguishable from the original files as possible to hopefully avoid possible breakage, my spontaneous idea was to take UUID V4... but wait, they're 36 characters long, so I'll just hash them with SHA and cut them down to 32 characters, hopefully not increasing the chance of collisions too much.

Except, you know...

1. UUIDs actually carry exactly 128 bits, which is what I needed in the first place...
2. ...and most embarrassingly, upon closer inspection, the `objectId`s in ForeFlight-generated files are all valid UUID V4 with the dashes removed :-)

All I can say is that I got too carried away in my hacking spree.